spyder/api/plugins.py

```diff
@@ -226,6 +227,20 @@ def refresh_actions(self):
         self.plugin_actions = self.get_plugin_actions() + additional_actions
         add_actions(self.options_menu, self.plugin_actions)
 
+        # Show icons in Mac plugin menus
+        if sys.platform == 'darwin':
+
+            self.options_menu.aboutToShow.connect(lambda plugin_actions=self.plugin_actions: self.verify_menu_actions(plugin_actions, True))
+            self.options_menu.aboutToHide.connect(lambda plugin_actions=self.plugin_actions: self.verify_menu_actions(plugin_actions, False))
+
```

Comment:
@juanis2112 let's move this conection to the `initialize_plugin` method and only connect the about to hide one

also, we do not need to use the parameter on the lambda.

something like:

```python
        # Show icons in Mac plugin menus
        if sys.platform == 'darwin':
            self.options_menu.aboutToHide.connect(
                lambda: self.verify_menu_actions(False))
```